### PR TITLE
Workaround for sr-13263

### PR DESCRIPTION
**Pull request by philipturner**
Summary: half-fix for `BatchNorm` compiler crash

On debug and release (only with -Onone), this compiles successfully. On fully optimized release, this still fails to compile and produces the stack trace mentioned in https://github.com/apple/swift/pull/38745#issuecomment-947890567. In short, I independently implemented your "number of patches".

This was tested on Google Colab (dual-core x64) with the 2021-11-12 toolchain. Later toolchains introduce other bugs that prevent compilation in RecurrentLayer  - https://github.com/tensorflow/swift-apis/pull/1184#issuecomment-1008006067.

Let's try keeping as much discussion as possible on https://github.com/tensorflow/swift-apis/issues/1189#issue-1097286651. A PR on a fork isn't as accessible as something created on tensorflow's repo.

### CODE REVIEW

**Comment by ProfFan** [comment]
Let me stash and merge.
